spec/rubocop/cop/rspec/repeated_example_group_body_spec.rb

```diff
@@ -0,0 +1,216 @@
+        it { cool_predicate_method }
+      end
+
+      RSpec.describe 'doing x' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated describe block body on line(s) [9]
+        it { cool_predicate_method }
+      end
+
+      context 'when a is true' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated context block body on line(s) [5]
+        it { cool_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'registers offense only for RSPEC namespace example groups in any order' do
+    expect_offense(<<-RUBY)
+      RSpec.describe 'doing x' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated describe block body on line(s) [5]
+        it { cool_predicate_method }
+      end
+
+      context 'when a is true' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated context block body on line(s) [1]
+        it { cool_predicate_method }
+      end
+
+      helpers.describe 'doing x' do
+        it { cool_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'registers offense only for example group' do
+    expect_offense(<<-RUBY)
+      RSpec.describe 'A' do
+        stub_all_http_calls()
+        allowed_statuses = %i[open submitted approved].freeze
+        before { create(:admin) }
+
+        describe '#load' do
+        ^^^^^^^^^^^^^^^^^^^ Repeated describe block body on line(s) [10]
+          it { cool_predicate_method }
+        end
+
+        describe '#load' do
+        ^^^^^^^^^^^^^^^^^^^ Repeated describe block body on line(s) [6]
+          it { cool_predicate_method }
+        end
+      end
+    RUBY
+  end
+
+  it 'skips `skip` and `pending` statements' do
+    expect_no_offenses(<<-RUBY)
+      context 'rejected' do
+        skip
+      end
+
+      context 'processed' do
+        skip
+      end
+
+      context 'processed' do
+        pending
+      end
+    RUBY
+  end
+
+  it 'registers offense correctly if example groups are separated' do
+    expect_offense(<<-RUBY)
+      describe 'repeated' do
+      ^^^^^^^^^^^^^^^^^^^^^^ Repeated describe block body on line(s) [7]
+        it { is_expected.to be_truthy }
+      end
+
+      before { do_something }
+
+      describe 'this is repeated' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated describe block body on line(s) [1]
+        it { is_expected.to be_truthy }
+      end
+    RUBY
+  end
+
+  it 'does not register offense for same examples with different data' do
+    expect_no_offenses(<<-RUBY)
+      context 'when admin' do
+        let(:user) { admin }
+        it { is_expected.to be_truthy }
+      end
+
+      context 'when regular user' do
+        let(:user) { staff }
+        it { is_expected.to be_truthy }
+      end
+    RUBY
+  end
+
+  it 'does not register offense for example groups without descriptions' do
+    expect_no_offenses(<<-RUBY)
+      context do
+        let(:preferences) { default_preferences }
+
+        it { is_expected.to eq false }
+      end
+
+      context do
+        let(:preferences) { %w[a] }
+
+        it { is_expected.to eq true }
```

Comment:
What if the examples were the same? Would it register the offense?

Reply:
My bad, actually this test should be in `RepeatedExampleGroupDescription` spec, but it is a good case for `RepeatedExampleGroupBody` too